subsys/net/ip/net_shell.c

```diff
@@ -6045,6 +6152,11 @@ SHELL_STATIC_SUBCMD_SET_CREATE(net_commands,
 	SHELL_CMD(websocket, NULL, "Print information about WebSocket "
 								"connections.",
 		  cmd_net_websocket),
+
+	#if defined(CONFIG_NET_DHCPV4)
+	SHELL_CMD(dhcp, &net_cmd_dhcp, "Show DHCP information.", cmd_net_dhcp),
+	#endif
+
```

Comment:
Hey

I did do this in order to reduce code footprint when functionality isn't available. When designing to flash constrained device. The .text section would grow if user don't want to enable dhcp itself in it's code but still use networking and networking shell. That's why I used an ifdef instead of showing the functionality isn't enabled at runtime.

I will correct to implement the architecture in the way you describe as doon as posible. Thank you by your feedback.

Reply:
I see your point. Thank you by explaining me the architecture of netshell.